lib/vector-core/src/config/id.rs

```diff
@@ -7,64 +7,32 @@ use std::{
     fmt,
 };
 
-#[derive(Debug, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
-pub enum ComponentScope {
-    Global,
-}
-
-impl fmt::Display for ComponentScope {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        match self {
-            Self::Global => write!(f, "global"),
-        }
-    }
-}
-
 #[derive(Debug, Clone, Eq, PartialEq, Hash)]
 pub struct ComponentKey {
     id: String,
-    scope: ComponentScope,
 }
 
 impl ComponentKey {
-    pub fn global<T: Into<String>>(id: T) -> Self {
-        Self {
-            id: id.into(),
-            scope: ComponentScope::Global,
-        }
-    }
-
     pub fn id(&self) -> &str {
-        self.id.as_str()
+        &self.id
     }
 
-    pub const fn scope(&self) -> &ComponentScope {
-        &self.scope
-    }
-
-    pub const fn is_global(&self) -> bool {
-        matches!(self.scope, ComponentScope::Global)
+    pub fn create_child(&self, name: &str) -> Self {
```

Comment:
We could also name this `join` to match the semantics of something like `Path`. Either works, just an idea.

Reply:
you're right, I like it better.